app/utils/transactionHelpers.ts

```diff
@@ -592,13 +613,36 @@ export function validateBakerStake(
     return undefined;
 }
 
+export function validateMemo(memo: string): string | undefined {
+    const asNumber = Number(memo);
+    if (
+        Number.isInteger(asNumber) &&
+        (asNumber > Number.MAX_SAFE_INTEGER ||
+            asNumber < Number.MIN_SAFE_INTEGER)
+    ) {
+        return `Only numbers up to ${Number.MAX_SAFE_INTEGER} is supported.`;
+    }
+    if (getEncodedSize(memo) > maxMemoSize) {
+        return `Memo is too large, encoded size must be at most ${maxMemoSize} bytes`;
+    }
+    // Check that the memo only contains ascii characters
+    // eslint-disable-next-line no-control-regex
+    if (/[^\u0000-\u007f]/.test(memo)) {
```

Comment:
This deserves its own utility function as it is a candidate for reuse. So refactor it into an `isASCII` method.